### DISCUSSION TITLE
Get original labels back

### DISCUSSION THREAD

**Discussion by AdarshMJ** [pyg-team/pytorch_geometric#4518] [Q&A]
Hi!

So I have graphs in the PyG format -
PyG graph DataBatch(x=[554, 11], edge_index=[2, 1320], edge_attr=[1320, 4], y=[31, 19], pos=[554, 3], idx=[31], name=[31], z=[554], batch=[554], ptr=[32])

I convert them to networkx graphs as follows -
`to_networkx(graph , node_attrs=['x'], graph_attrs=['y']))` 

On these NetworkX graphs I have to perform some manipulations, like adding edge/deletions based on some criteria and save them back to PyG graphs. But on doing this, I am not able to get the original labels back, basically I would want to be able to work on the modified graph using some GNN. 

After manipulating the graph if I convert them back to PyG graphs this is what I get -

`Data(edge_index=[2, 1322], weight=[1322], num_nodes=554)` 

The original information is lost. Is there any way to get the original labels back? Any insights are appreciated. 

Thank you!

**Comment by rusty1s** [comment]
Thanks for reporting. Do you have a small example to reproduce? My guess here is that your transformations on the `networkx` graph will remove any `data` attached to it. Does `list(G.nodes(data=True))` still hold information about `x`?

**Reply by AdarshMJ** [reply]
Hi! Thank you for your response.

I don't have a small example to reproduce, but I can share the Colab file. This is the error I get for 

`list(G.nodes(data=True))`
![Screenshot 2022-04-23 at 10 48 07](https://user-images.githubusercontent.com/24205200/164887530-4ba436a1-e65f-4ba4-8740-a79ab6babd27.png)

Im using this code - https://github.com/abojchevski/node_embedding_attack to add just a single edge to the graph, all the manipulations for this requires the graph to be in networkx/scipy_sparse matrix format. After introducing the edge I want to convert it back to PyG to train.

**Reply by rusty1s** [reply]
Can you call `list(G.nodes(data=True))` on the `networkx` graph (after doing your modifications and before calling `from_networkx`) rather than on the PyG `data` object?

**Reply by AdarshMJ** [reply]
Hi Yes, this is the output I get -
![Screenshot 2022-04-25 at 11 39 23](https://user-images.githubusercontent.com/24205200/165063357-a6c55383-04ce-4f2e-a485-b359e632a453.png)

**Reply by rusty1s** [reply]
Then this looks like a problem within your `networkx` routine. If the `networkx` graph loses track of node features, then PyG cannot add them via `from_networkx` as well.

**Reply by AdarshMJ** [reply]
Oh okay. I probably have to rethink about the method Im using to introduce edges. Thank you!
